src/librustc/mir/interpret/value.rs

```diff
@@ -332,6 +332,13 @@ impl<'tcx, Tag> Scalar<Tag> {
         Ok(b as u8)
     }
 
+    pub fn to_u16(self) -> EvalResult<'static, u16> {
+        let sz = Size::from_bits(16);
+        let b = self.to_bits(sz)?;
+        assert_eq!(b as u16 as u128, b);
```

Comment:
Why is this assertion necessary? I also think that `from_bits` or `to_bits` should probably be doing the cast to the right type? Or am I missing out on something?

Reply:
It's copied from the other accessors already in Rustc.

Reply:
(They had existing implementations for all numeric types apart from u16, u128, i16, i128)

Reply:
It feels a bit icky, and on some platforms (which have slow `u128` support) also quite slow. Maybe it's not worth worrying about though.

Reply:
I don't know. I was just following the precedent:
https://github.com/rust-lang/rust/blob/9f8396158457943b63bb39534b9ef86868f7cfac/src/librustc/mir/interpret/value.rs#L339-L395

Reply:
OK, let's leave it as-is.